dipy/segment/clustering.py

```diff
@@ -657,3 +658,63 @@ def _traverse(node, level=0):
 
         _traverse(self.root)
         return clusters
+
+
+def qbx_with_merge(streamlines, thresholds,
+                   nb_pts=20, select_randomly=None, verbose=True):
+    if verbose:
+        t = time()
+    len_s = len(streamlines)
+    if select_randomly is None:
+        select_randomly = len_s
+    indices = np.random.choice(len_s, min(select_randomly, len_s),
```

Comment:
Personally, I prefer using a dedicated random generator instead of relying on `np.random`. It gives you more control for reproducibility.

I would add a new optional argument `rng=None` to the function and do
```python
indices = np.arange(len(streamlines))
if select_randomly:
    rng = np.random.RandomState(None) if rng is None else rng
    rng.shuffle(indices)
```